lucille-core/src/main/java/com/kmwllc/lucille/stage/BuildNested.java

```diff
@@ -0,0 +1,332 @@
+      }
+    }
+
+    if (numObjects != null && (minNumObjects != null || maxNumObjects != null)) {
+      throw new StageException("Specify either num_objects or (min_num_objects & max_num_objects), not both.");
+    }
+
+    this.parsedEntries = Collections.unmodifiableMap(parseEntries(entriesCfg));
+  }
+
+  // Start generator stages
+  @Override
+  public void start() throws StageException {
+    if (generatorsConfig == null) {
+      return;
+    }
+
+    // Iterate over each generator param
+    for (String key : generatorsConfig.root().keySet()) {
+      Config sub = generatorsConfig.getConfig(key);
+      if (!sub.hasPath("class")) {
+        throw new StageException("generators." + key + " must include a 'class' property");
+      }
+
+      // Inline default temp field name
+      String tmpField = sub.hasPath("field_name") ? sub.getString("field_name") : ".bn_gen." + key;
+
+      // Add default name and field name
+      Config injected = sub;
+      if (!sub.hasPath("field_name")) {
+        injected = ConfigFactory.parseMap(Map.of(
+            "name", "bn_gen_" + key,
+            "field_name", tmpField
+        )).withFallback(sub);
+      }
+
+      // Create generator
+      Stage gen = instantiateStage(injected);
+      gen.start();
+      generators.put(key, gen);
+      genOutField.put(key, tmpField);
+    }
+  }
+
+  @Override
+  public void stop() throws StageException {
+    for (Stage generator : generators.values()) {
+      try {
+        generator.stop();
+      } catch (Exception ignored) {}
+    }
+  }
+
+  // Parse entries to split paths at .
+  private static Map<List<String>, String> parseEntries(Config entriesCfg) throws StageException {
+    Map<List<String>, String> out = new LinkedHashMap<>(entriesCfg.root().size());
+    Map<String, Object> flat = new LinkedHashMap<>(entriesCfg.root().unwrapped());
+
+    for (Map.Entry<String, Object> ent : flat.entrySet()) {
+      String dest = ent.getKey();
+      String src  = ent.getValue() == null ? "" : String.valueOf(ent.getValue());
+      String trimmedDest = dest == null ? "" : dest.trim();
+      String trimmedSrc = src == null ? "" : src.trim();
+
+      if (trimmedDest.isEmpty()) {
+        throw new StageException("Invalid mapping (empty destination path).");
+      }
+
+      String[] parts = trimmedDest.split("\\.");
+      for (String p : parts) {
+        if (p.isEmpty()) {
+          throw new StageException("Invalid destination '" + trimmedDest + "' (empty segment).");
+        }
+      }
+
+      List<String> key = List.of(parts);
+      if (out.containsKey(key)) {
+        throw new StageException("Duplicate destination '" + dest + "'.");
+      }
+
+      out.put(key, trimmedSrc);
+    }
+
+    return out;
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    // Get doc as map so we can use previous field values
+    Map<String, Object> map = doc.asMap();
```

Comment:
can you give an example of why this is necessary?
where would be looking for a value in this map that we can't get directly from the doc?